src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -376,12 +379,50 @@ protected SCMRevision doRetrieve(SCMHead head, TaskListener listener, GHReposito
         if (head instanceof PullRequestSCMHead) {
             int number = ((PullRequestSCMHead) head).getNumber();
             ref = repo.getRef("pull/" + number + "/merge");
+            // TODO if we already had the GHPullRequest.user.login in a field in the PullRequestSCMHead, we could pass that directly and save an API call
+            String trustedBase = trustedReplacement(repo, repo.getPullRequest(number));
+            if (trustedBase != null) {
+                return new UntrustedPullRequestSCMRevision(head, ref.getObject().getSha(), trustedBase);
+            }
         } else {
             ref = repo.getRef("heads/" + head.getName());
         }
         return new SCMRevisionImpl(head, ref.getObject().getSha());
     }
 
+    @Override
+    public SCMRevision getTrustedRevision(SCMRevision revision, TaskListener listener) throws IOException, InterruptedException {
+        if (revision instanceof UntrustedPullRequestSCMRevision) {
+            PullRequestSCMHead head = (PullRequestSCMHead) revision.getHead();
+            UntrustedPullRequestSCMRevision rev = (UntrustedPullRequestSCMRevision) revision;
+            listener.getLogger().println("Loading trusted files from target branch at " + rev.baseHash + " rather than " + rev.getHash());
```

Comment:
> So, the 'synthetic' merge commit, is this something that points to a revision with the 'trusted' files replaced on it somehow?

No, this exists prior to this PR. The comment about the synthetic commit was more for @recena’s benefit; my proposal to fix the way this plugin represents revisions of PRs would involve changes to some of the same code, so it is blocked by this PR in that sense.

> I seem to be missing the key part of these 3 PRs that somehow specifies the Jenkinsfile from the trusted branch is the only thing used from that source, rather than the entire commit.

Done [here](https://github.com/jenkinsci/workflow-plugin/pull/244/files#diff-94b5a667704d0337bc1dad26d12b38caL80). Compare [this](https://github.com/jglick/workflow-plugin/blob/421efc44ba8a6274d48de6b62174af978b342ba4/multibranch/src/main/java/org/jenkinsci/plugins/workflow/multibranch/SCMVar.java#L88), used for `checkout scm`, which remains unmodified by the PR.